internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -430,7 +436,10 @@ func (m DeploymentResource) Create() sdk.ResourceFunc {
 				prop.NetworkProfile.NetworkInterfaceConfiguration.SubnetId = pointer.FromString(model.NetworkInterface[0].SubnetId)
 			}
 
```

Comment:
should we add a check that returns an error if the plan is basic and the user attempts to set these properties?

Reply:
I considered that, but I don't think so. There's server side validation to fail fast with 400 Bad Request and a useful error message if the user attempts something like `plan = "basic"` and `capacity = 30`.

There are many server-side validations for the `nginxdeployment.NginxDeployment{}` data, and I didn't want to start duplicating that logic in the terraform provider. NGINXaaS evolves fairly rapidly, and I wanted to avoid bad UX coming from inconsistent validation logic if `azurerm_nginx_deployment` follows different rules than the azure CLI or portal.

Does that sound reasonable?

Reply:
@katbyte please let me know what you think, I'd love to get this merged.

Reply:
@katbyte thoughts? I can add a check if you feel strongly about it.

Reply:
Well this PR is due to the server side validation code not handling this in a good way thou eh?

I would rather us check for bad data and return an error to the user over us silently not sending the data to the API that triggers a bad data response (which lacks a clear action for the user to take i presume) Not to mention if the user sets the value to 20 it will just silently not set it and also not trigger and server side validation.

so could we please change this to returning an error with basic and a capacity above zero rather then just omitting it? you might also have to change this to check if the property is set woth d.GetOkExists()

Reply:
> Well this PR is due to the server side validation code not handling this in a good way thou eh?

Not quite. This PR is due to `azurerm_nginx_deployment` applying a default capacity of 20, with no way for the user to specify "no capacity". The new "Basic" SKU does not support scaling of any kind, but `azurerm_nginx_deployment` was always sending a `capacity`.

> I would rather us check for bad data and return an error to the user over us silently not sending the data to the API that triggers a bad data response (which lacks a clear action for the user to take i presume) Not to mention if the user sets the value to 20 it will just silently not set it and also not trigger and server side validation.

Agreed. The problem is there's no mechanism for a user to provide _good_ `capacity` data for a basic SKU

```hcl
resource "azurerm_nginx_deployment" "example" {
  sku  = "basic"
  # implicity gets capacity = 20, which is invalid
}
```

This weird conditional is an attempt to get TF to _stop_ sending a capacity when the user doesn't ask for one. We currently have our server-side validation hacked to ignore `capacity` when for basic SKUs, but we want to remove that and return 400s with a good error message, like we do for other invalid combinations.

> so could we please change this to returning an error with basic and a capacity above zero rather then just omitting it? you might also have to change this to check if the property is set woth d.GetOkExists()

Unfortunately due to the `Default` value, the capacity is always above zero regardless of what the user does.

I tried `GetOkExists`, but it still returns `true` when I omit `capacity`. Per the doc string, `GetOkExists` doesn't seem to work when there's a default value.

I didn't love the solution in this PR, it seemed like the least bad option that remained backward compatible. We considered a few alternatives:

1. remove the `Default: 20` - this would be a backwards incompatible change for TF users, and we didn't want to be that disruptive nor wait for the 4.0 release
2. change the `Capacity` `ValidationFunc` to `validation.IntAtLeast(0)` and add docs saying you have to use `capacity = 0` for basic SKUs - this felt like an extra step for users, and we were worried about allowing zeros into peoples' TF state setting up more backward compat problems in the future

Based on you're feedback, it sounds like the `IntAtLeast(0)` approach is preferable. I can update this PR to match, do you see any other options that might be better?

Reply:
Its worth noting the 4.0 release will be within the next month or so so that could be changed, it might make sense to just have it be computed without a default as i presume the API will set a default of 20? and then GetOkExists could work.

outside that as it is not supported with basic the check could simply be `if model.Capacity > 0 && ! strings.HasPrefix(model.Sku, "basic") {`  ? with a error thrown if capacity is not 20. its not ideal as a user could specify a capacity and not realize its not being sent but that could be resolved later on as i think with framework we will have the ability to know if its not set.

Reply:
updated!